tests/test-update-cleanup-workaround.c

```diff
@@ -0,0 +1,213 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*-
+ *
+ * Copyright © 2017 Kinvolk GmbH
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * Authors:
+ *  - Krzesimir Nowak <krzesimir@kinvolk.io>
+ */
+
+#include "misc-utils.h"
+#include "spawn-utils.h"
+#include "ostree-spawn.h"
+#include "eos-test-utils.h"
+#include "eos-test-convenience.h"
+
+#include <gio/gio.h>
+#include <locale.h>
+#include <string.h>
+
+static void
+save_old_deployment_commit_in_ref (EtcData *data,
+                                   const gchar *ref)
+{
+  g_auto(GStrv) deployment_ids = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree gchar *old_commit_id = NULL;
+  gchar *dot_ptr;
+  g_autoptr(GFile) client_repo = NULL;
+  g_auto(CmdResult) ref_created = CMD_RESULT_CLEARED;
+
+  g_assert_nonnull (data);
+  g_assert_nonnull (data->client);
+  g_assert_nonnull (ref);
+
+  eos_test_client_get_deployments (data->client,
+                                   default_remote_name,
+                                   &deployment_ids,
+                                   &error);
+  g_assert_no_error (error);
+  g_assert_cmpint (g_strv_length (deployment_ids), ==, 2);
+
+  old_commit_id = g_strdup (deployment_ids[1]);
```

Comment:
OK. If you could add that as a brief comment (like `/* Index 1 is always guaranteed to be the old deployment. */`) that would be great.

If you could improve the upstream ostree documentation on this, that would be even better, but don’t treat that as a priority. In any case, please file an issue on the upstream github for it. Thanks.